### PR TITLE
23.06 Install Page Update

### PR DESCRIPTION
**Pull request by jarmak-nv**
This turned out smaller than expected, probably could have been part of #394 but I like smaller PRs anyway. 

- Removed the mention of `Python 3.8` support in pip installation guide
- Added CUDA 12.0 as a supported CUDA
- Added a note that the Docker/Conda CUDA 11 installations work fine on CUDA 12 machines, but that pip needs to match

Contributes to #386

### CODE REVIEW

**Comment by jarmak-nv** [comment]
@bdice Quick question on the bolded part below for [this line](https://github.com/rapidsai/docs/pull/396/files#diff-b1b679a811f38c824cce215c29f17a75472d763031023c125d190e579f723b34R309) since I think you're the person to ask (apologies if not!):
> Follow [this helpful developer guide](https://docs.nvidia.com/cuda/wsl-user-guide/index.html#cuda-support-for-wsl2){: target="_blank"} and then [install the CUDA Toolkit without drivers](https://developer.nvidia.com/cuda-11-8-0-download-archive?target_os=Linux&target_arch=x86_64&Distribution=WSL-Ubuntu&target_version=2.0&target_type=deb_local){: target="_blank"} into the WSL2 instance. **It's important to execute `sudo apt-get -y install cuda-toolkit` instead of `sudo apt-get -y install cuda` to avoid installing a GPU driver into WSL2. The Windows host system provides the driver to WSL2.**

If they follow the steps from the link, is this accurate? We link to the guide to do a local install of a specific WSL Ubuntu 11.8 CTK that does not include the drivers. 

The guide says to end with the command `sudo apt-get -y install cuda` rather than `cuda-toolkit`. If that's wrong, we should reach out to get that fixed too.

Relevant since I'll need to add the CUDA 12 link and just want to make sure it's correct. FWIW when I installed it on my WSL2 machine a couple months ago I followed the guide exactly without issue.

**Comment by bdice** [comment]
@jarmak-nv I want to be sure that users don't get tripped up by the second link. It says: 

https://developer.nvidia.com/cuda-11-8-0-download-archive?target_os=Linux&target_arch=x86_64&Distribution=WSL-Ubuntu&target_version=2.0&target_type=deb_local

image

Note the last line, `sudo apt-get -y install cuda` is incorrect and should be `cuda-toolkit`.

**Comment by bdice** [comment]
> If that's wrong, we should reach out to get that fixed too.

I'm not sure how to move that forward -- but I think the download page should never advise installing a driver into a WSL system. If you know where to ask, please inquire about that and feel free to tag me!

**Comment by jarmak-nv** [comment]
Don't the previous steps in the instructions make it install the version they downloaded though? the `cuda-repo-wsl-ubuntu-11-8-local_11.8.0-1_amd64.deb` (and any `WSL-Ubuntu` version) doesn't include a driver.

Per [NV docs](https://docs.nvidia.com/cuda/wsl-user-guide/index.html#cuda-support-for-wsl-2)
> We recommend developers to use a separate CUDA Toolkit for WSL 2 (Ubuntu) available from the [CUDA Toolkit Downloads](https://developer.nvidia.com/cuda-downloads?target_os=Linux&target_arch=x86_64&Distribution=WSL-Ubuntu&target_version=2.0) page to avoid this overwriting. This WSL-Ubuntu CUDA toolkit installer will not overwrite the NVIDIA driver that was already mapped into the WSL 2 environment.

I could easily be missing something though!

**Comment by bdice** [comment]
@jarmak-nv Thanks for pointing that out. I don't think I was aware of that difference until now. Maybe I'm concerned about something that has already been solved? I know that users (and my former self) have run into situations where drivers were installed in WSL but maybe it's a case where they (and I) haven't read the directions carefully enough.

**Comment by jarmak-nv** [comment]
Just to close the loop on this, I nuked one of my WSL Ubuntu 22.04 installs and started fresh following the page instructions and then doing a pip install of cuDF. 

Our suggested test command of
```
 import cudf
 print(cudf.Series([1, 2, 3]))
```
So I'll remove the note when I add CUDA 12 on Monday because I think it could lead to extra confusion and it's not needed.

File: install/install.md
Comment (bdice) [inline install/install.md:62]: Can we verify this? This will install packages that match the driver and not the toolkit, if I understand the behavior of `nvidia-smi` correctly. I believe if you install CUDA 12 drivers (e.g. 525) and a CUDA 11 toolkit, this will install an incompatible version of RAPIDS (it'll install `-cu12` but you need to match the toolkit, `-cu11`). You can test this on some of the ASE lab machines, which have driver 525 and CUDA 11.8 toolkit installed.

**Review by ajschmidt8 — APPROVED** [review]

File: install/install.md
Comment (bdice) [inline install/install.md:126]: Fix URL.
```suggestion
- <i class="fas fa-check-circle"></i> [CUDA 12.0](https://developer.nvidia.com/cuda-12-0-1-download-archive){: target="_blank"} with Driver 525.60.13 or newer **for pip installations only**
```

File: install/install.md
Comment (bdice) [inline install/install.md:130]: What does "CUDA 12.x machine" mean? We need to speak in terms of **driver** or **toolkit**. "Machine" is too ambiguous.

Comment (jarmak-nv) [reply install/install.md:130]: Agreed! 

>**Note**: RAPIDS Docker and Conda CUDA 11.x environments function as expected on a system with a CUDA 12.x driver and they include their own toolkit pieces. <br>pip installations require using the matching wheel to the system's installed CUDA toolkit. For CUDA 11.x toolkits, install the <code>-cu11</code> wheels, and for CUDA 12.x tookits install the <code>-cu12</code> wheels. If your installation has a CUDA 12.x driver but a CUDA 11.x toolkit, use the <code>-cu11</code> wheels.

?

File: install/install.md
Comment (bdice) [reply install/install.md:62]: Yeah... I wouldn't rely on CUDA being installed to `/usr/local`. Can we just say "install the version that matches your CUDA toolkit version" and not attempt to write user-proof code to do it automatically?

Comment (bdice) [reply install/install.md:62]: (For a counter-case of the `/usr/local` code, you could install the CUDA toolkit via conda and not install a system CTK, only a driver.)

File: install/install.md
Comment (bdice) [reply install/install.md:130]: > and they include their own toolkit pieces.

change to
> because they include their own toolkit.

File: install/install.md
Comment (jarmak-nv) [reply install/install.md:62]: Yup, agree we should just leave it to the user to match them. If we get lots of issues filed due to challenges here we can reconsider.

File: install/install.md
Comment (jarmak-nv) [reply install/install.md]: I've made these changes 🎉

File: install/install.md
Comment (bdice) [inline install/install.md:62]: Avoid smart (curly) quotes.
```suggestion
<i class="fas fa-info-circle"></i> pip installations require using the matching wheel to the system's installed CUDA toolkit. For CUDA 11 toolkits, install the `-cu11` wheels, and for CUDA 12 tookits install the `-cu12` wheels. If your installation has a CUDA 12 driver but a CUDA 11 toolkit, use the `-cu11` wheels. <br/>
```

File: install/install.md
Comment (bdice) [inline install/install.md:126]:
```suggestion
 **Note**: RAPIDS Docker images and conda packages for CUDA 11 can be used on a system with a CUDA 12 driver because they include their own CUDA toolkit. pip installations require using a wheel matching the system's installed CUDA toolkit. For CUDA 11 toolkits, install the <code>-cu11</code> wheels, and for CUDA 12 toolkits install the <code>-cu12</code> wheels. If your installation has a CUDA 12 driver but a CUDA 11 toolkit, use the <code>-cu11</code> wheels.
```

File: install/install.md
Comment (bdice) [inline install/install.md:242]: I re-ordered this sentence a bit. The "Beginning with..." made sense when pip packages were new, but not so much now that they're stable.

I googled around a bit and it seems like "NVIDIA Python Package Index" is the name that is being used more broadly, not "NVIDIA Index."
```suggestion
RAPIDS 23.06 pip packages of cuDF, dask-cuDF, cuML, cuGraph, cuSpatial, RMM, and RAFT are available for CUDA 11 and CUDA 12 on the NVIDIA Python Package Index.
```
Also, do we need to list cuCIM? Are we missing anything else?

File: install/install.md
Comment (bdice) [inline install/install.md:318]:
```suggestion
	- The installed CUDA Toolkit version must match the pip wheel version (`-cu11` or `-cu12`)
```

**Review by bdice — APPROVED** [review]
I proposed some changes. I'll approve to unblock this PR. Please resolve my suggestions as you see fit.

File: install/install.md
Comment (jarmak-nv) [reply install/install.md:242]: cuCIM is on pypi proper, this should be correct list for what's on pypi.nvidia.com unless we want to include ptxcompiler or something. I wanted to only include items on the selector though. Which makes me realize I need to add a button to the selector for `dask-cudf`.

**Comment by jarmak-nv** [comment]
I like the changes - technical writing like this always takes two sets of eyes at a minimum so thank you!

**Pull request merged**
